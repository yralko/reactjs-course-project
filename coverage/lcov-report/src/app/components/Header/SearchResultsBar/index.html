<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/app/components/Header/SearchResultsBar</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\..\index.html">All files</a> src/app/components/Header/SearchResultsBar
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">52.63% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>10/19</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">56.25% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>9/16</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">42.86% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/8</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="SearchResultsBar.js"><a href="SearchResultsBar.js.html">SearchResultsBar.js</a></td>
	<td data-value="43.75" class="pic low"><div class="chart"><div class="cover-fill" style="width: 43%;"></div><div class="cover-empty" style="width:57%;"></div></div></td>
	<td data-value="43.75" class="pct low">43.75%</td>
	<td data-value="16" class="abs low">7/16</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="12" class="abs medium">6/12</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="6" class="abs low">2/6</td>
	<td data-value="71.43" class="pct medium">71.43%</td>
	<td data-value="7" class="abs medium">5/7</td>
	</tr>

<tr>
	<td class="file empty" data-value="index.css"><a href="index.css.html">index.css</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file high" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue May 08 2018 18:10:44 GMT+0300 (Belarus Standard Time)
</div>
</div>
<script src="..\..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\..\sorter.js"></script>
<script src="..\..\..\..\..\block-navigation.js"></script>
</body>
</html>
