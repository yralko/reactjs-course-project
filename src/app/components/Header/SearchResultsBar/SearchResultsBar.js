import React from 'react';
import { connect } from 'react-redux';
import * as actions from '../../../store/actions';
import classes from './index.css';

const SearchResultsBar = props => (
  <div className={classes.SearchResultsBar}>
    <div className="total-found">
      <span>{props.foundFilms.length} {props.foundFilms.length === 1 ? 'film' : 'films'} found</span>
    </div>
    <div className="sort-options">
      <span>Sort by </span>
      <button
        className={classes.sortOptions}
        onClick={() => props.sortFilms('release_date')}
      >
        release date
      </button>
      <button
        className={classes.sortOptions}
        onClick={() => props.sortFilms('vote_average')}
      >
        rating
      </button>
    </div>
  </div>
);

const mapStateToProps = (state) => {
  return {
    foundFilms: state.foundFilms,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    sortFilms: sortParameter => dispatch(actions.sortFilms(sortParameter)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(SearchResultsBar);
