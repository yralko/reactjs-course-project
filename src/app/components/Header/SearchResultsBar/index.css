.SearchResultsBar {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 0 auto;
  padding: 5px 15%;
  background-color: #eeeeee;
}

.sortOptions {
  display: inline-block;
  padding: 5px;
  margin: 0 5px;
  color: black;
  border: none;
  background-color: transparent;
}

.sortOptions.active {
  color: red;
}

.sortOptions:hover {
  cursor: pointer;
}
