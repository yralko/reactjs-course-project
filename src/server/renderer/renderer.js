import React from 'react';
import { renderToString } from 'react-dom/server';
import { Provider } from 'react-redux';
import { StaticRouter as Router } from 'react-router-dom';
import App from '../../app/App';


export default (req, store, context) => {
  const root = (
    <Provider store={store}>
      <Router location={req.url} context={context}>
        <App />
      </Router>
    </Provider>
  );

  const content = renderToString(root); 

  return `
  <!doctype html>
  <html lang="en">
  <head>
    <meta charset="utf-8">
    <title>The HTML5 Herald</title>
    <base href="/" />
    <link rel="stylesheet" href="style.css" />
  </head>
  <body>
    <div id="root">${content}</div>
    <script>window.INITIAL_STATE = ${JSON.stringify(store.getState())}</script>
    <script src="bundle.js"></script>
  </body>
  </html>
  `;
};
