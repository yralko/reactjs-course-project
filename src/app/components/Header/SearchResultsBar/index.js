import SearchResultsBar from './SearchResultsBar';

export default SearchResultsBar;
