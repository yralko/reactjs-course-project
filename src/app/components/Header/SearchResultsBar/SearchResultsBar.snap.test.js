import React from 'react';
import renderer from 'react-test-renderer';
import { SearchResultsBar } from './SearchResultsBar';

test('matches snapshot', () => {
  const mockState = {
    foundFilms: [
      { title: 'film1' },
      { title: 'film2' },
      { title: 'film3' },
    ],
  };

  const tree = renderer.create(<SearchResultsBar foundFilms={mockState.foundFilms} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
