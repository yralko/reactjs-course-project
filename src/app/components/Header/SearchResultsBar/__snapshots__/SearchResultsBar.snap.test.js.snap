// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className={undefined}
>
  <div
    className="total-found"
  >
    <span>
      3
       
      films
       found
    </span>
  </div>
  <div
    className="classes.sort-options"
  >
    <span>
      Sort by 
    </span>
    <button
      className={undefined}
      onClick={[Function]}
    >
      release date
    </button>
    <button
      className={undefined}
      onClick={[Function]}
    >
      rating
    </button>
  </div>
</div>
`;
